### ISSUE TITLE
Render Error Reporting

### ISSUE THREAD

**Issue by tomgallagher** [diegomura/react-pdf#893]
First, can I say thanks for this package, it's way better than any pdf generator I've used before.

I've come up against a problem and I was hoping for some pointers.

I'm using version 2.0.0-beta.5, which has proved to be very helpful with the new features, especially `minPresenceAhead`.

I've just come up against this message at the render stage:

> TypeError: Cannot read property 'height' of undefined
>     at setNodeHeight (resolveDimensions.js:72)
>     at resolveDimensions.js:85
>     at tap (tap.js:26)
>     at _dispatchable.js:38
>     at _curry2.js:19
>     at f1 (_curry1.js:16)
>     at _pipe.js:3
>     at _pipe.js:3
>     at _pipe.js:3
>     at _arity.js:10
>     at resolveDimensions.js:176

What I would really like here is a bit more information in the error, for example some info about which node is causing a problem.

In the meantime, could you give me any pointers about what kind of mistake on my part could have caused this? I've been using the package constantly for the last 3 weeks and this is the first time I've encountered this message.

I'm going to try to find the error but it's a massive document and a long job to wade through it.

Thanks!

Tom

**Comment by tomgallagher** [comment]
Found the problem, so all good. Thought I'd leave this in case you thought it was worth considering.

**Comment by richcsmith** [comment]
@tomgallagher I'm also using 2.0.0-beta.5 and running into the same issue trying to load an image from a blob URL. Mind sharing how you resolved it?

**Comment by tomgallagher** [comment]
Yes, it was my problem, not the package. This issue is a request to have the renderer tell us where we had made the mistake. Otherwise you may have a big document where you have to go around removing one bit at a time until you see what's caused the error.

In your case, however, you know it's the image that's the problem. I've never tried to load an image from a blob url so I don't know if that's acceptable. I've only used remote urls and data-uri.

**Comment by tomgallagher** [comment]
Check out the image resolver code in `src/utils/image.js`, it's not very complex.

```
const fetchRemoteFile = async (uri, options) => {
  const response = await fetch(uri, options);

  const buffer = await (response.buffer
    ? response.buffer()
    : response.arrayBuffer());

  return buffer.constructor.name === 'Buffer' ? buffer : Buffer.from(buffer);
};
```

**Comment by richcsmith** [comment]
Sorry for the late response—I'm still working through this and haven't figured out a solution. As it turns out, image rendering wasn't the source of the issue. This is the error I'm seeing at render time (similar to you):
```
TypeError: Cannot read property 'height' of undefined
    at setNodeHeight (react-pdf.browser.es.js?e5f2:6186)
    at eval (react-pdf.browser.es.js?e5f2:6197)
    at tap (tap.js?863b:26)
    at eval (_dispatchable.js?da3b:38)
    at eval (_curry2.js?8c6c:19)
    at f1 (_curry1.js?a797:16)
    at eval (_pipe.js?a8f3:3)
    at eval (_pipe.js?a8f3:3)
    at eval (_pipe.js?a8f3:3)
    at eval (_arity.js?833a:10)
```

I understand that this issue is to ask for more descriptive error messages, but what I'm asking you is what exactly was the issue in your code that caused this error in the first place?

**Comment by idl3** [comment]
I can confirm that the above suggestion to use a ternary operation to `null` instead of `&&` solved my issue with `TypeError: Cannot read property 'height' of undefined` as well.

**Comment by bobber205** [comment]
> Thank you for your reply, and apologies for the late response—I'm still working through this and haven't figured out a solution. As it turns out, image rendering wasn't the source of the issue. This is the error I'm seeing at render time (similar to you):
> 
> ```
> TypeError: Cannot read property 'height' of undefined
>     at setNodeHeight (react-pdf.browser.es.js?e5f2:6186)
>     at eval (react-pdf.browser.es.js?e5f2:6197)
>     at tap (tap.js?863b:26)
>     at eval (_dispatchable.js?da3b:38)
>     at eval (_curry2.js?8c6c:19)
>     at f1 (_curry1.js?a797:16)
>     at eval (_pipe.js?a8f3:3)
>     at eval (_pipe.js?a8f3:3)
>     at eval (_pipe.js?a8f3:3)
>     at eval (_arity.js?833a:10)
> ```
> 
> I understand that this issue is to ask for more descriptive error messages, but what I'm asking you is what exactly was the issue in your code that caused this error in the first place?
> 
> **Edit:** I actually just figured it out about 5 minutes after I posted this. I was conditionally rendering components using a double ampersand (`&&`), which breaks something in the Yoga renderer.
> 
> In case anyone else runs into this issue, here's what worked for me when rendering components:
> 
> ```
> // Do this
> return displayComponent ? <View /> : null;
> 
> // Don't do this
> return displayComponent && <View />;
> ```

Do you have any proof the `&&` is an issue with the Yoga renderer? I've been googling it for awhile and while admittedly googling an operator like an double ampersand is hard for google to parse but I still can't find any evidence of that being a known issue

**Comment by richcsmith** [comment]
Sorry, I don't have any proof. My memory of the troubleshooting I did is a little fuzzy, but I seem to remember digging into the stack trace to determine that it was an issue with Yoga. Are you also running into this?

**Comment by bobber205** [comment]
Not really just curious -- I just reported this https://github.com/diegomura/react-pdf/issues/1513 any bells rang by it? <3
